packages/block-editor/src/components/block-switcher/index.js

```diff
@@ -219,13 +224,16 @@ export const BlockSwitcher = ( { clientIds, disabled, isUsingBindings } ) => {
 			const blockType = getBlockType( firstBlockName );
 
 			let _icon;
+			let _hasTemplateLock;
 			if ( _isSingleBlockSelected ) {
 				const match = getActiveBlockVariation(
 					firstBlockName,
 					getBlockAttributes( clientIds[ 0 ] )
 				);
 				// Take into account active block variations.
 				_icon = match?.icon || blockType.icon;
+				_hasTemplateLock =
```

Comment:
TODO:

This is not yet correct.  We'll need to check all selected blocks for `contentOnly`.

My guess is that if a single block among multiple select is locked, then we can't show the transformation options. 

The other option is to show them and (try to) only apply the transformation to the non-locked blocks.

Reply:
It's in a branch with the condition `_isSingleBlockSelected`, so should be ok? Or was the comment meant to be on the lines below?

Most of the transforms for multiple blocks are grouping actions, so they're valid, but I think it might still be best to not show them if a content only locked block is selected. These are already groups, so it feels confusing. I'm personally in favor of what you have.

It's something that can be adjusted in future if needed, but my guess is most users wouldn't even find that transform menu anyway.

Reply:
> Or was the comment meant to be on the lines below?

It was for a previous commit. Thanks for double checking!